test-case-reporting/test/test_result_record.test.ts

```diff
@@ -84,294 +84,329 @@ describe('TestResultRecord', () => {
+          },
+          {
+            id: 'e3c1257d76c0b4d0f0c1307161ab5424',
+            name:
+              'when the moon hits your eye | when i was a young boy | when the fires come | it skips',
+          }
+        );
+
+        expect(testCases).toMatchObject(sampleTestCases);
+      });
 
-    it('inserts test cases', async () => {
-      await testResultRecord.storeTravisReport(sampleTravisReport);
-
-      const testCases: Array<DB.TestCase> = await db<DB.TestCase>(
-        'test_cases'
-      ).select();
-
-      const sampleTestCases: Array<DB.TestCase> = [
-        {
-          id: '8fd7659c797d5b46f64917937d4805f9',
-          name: 'when test is good | it passes',
-        },
-        {
-          id: '8a3d71d66b2913bb981a8d4f2a2930db',
-          name: 'when test is bad | it fails',
-        },
-        {
-          id: 'c5cf7c15d50ec660c3b10b6c91bfe3f8',
-          name: 'when test was skipped | it skipped',
-        },
-        {
-          id: '36340965686c32694f88f06c6a3f71ac',
-          name: '🤖 when passing test has emojis 🤖 | it passes 🎉',
-        },
-        {
-          id: 'e3c1257d76c0b4d0f0c1307161ab5424',
-          name:
-            'when the moon hits your eye | when i was a young boy | when the fires come | it skips',
-        },
-      ];
-
-      expect(testCases).toMatchObject(sampleTestCases);
+      it('inserts test runs', async () => {
+        await testResultRecord.storeTravisReport(sampleTravisReport);
+
+        const testRuns: Array<DB.TestRun> = await db<DB.TestRun>(
+          'test_runs'
+        ).select();
+
+        const sampleTestRuns: Array<DB.TestRun> = [
+          {
+            'test_case_id': '8fd7659c797d5b46f64917937d4805f9',
+            status: 'PASS',
+            'duration_ms': 42,
+            'job_id': 1,
+          },
+          {
+            'test_case_id': '8a3d71d66b2913bb981a8d4f2a2930db',
+            status: 'FAIL',
+            'duration_ms': 1337,
+            'job_id': 1,
+          },
+          {
+            'test_case_id': 'c5cf7c15d50ec660c3b10b6c91bfe3f8',
+            status: 'SKIP',
+            'duration_ms': 413,
+            'job_id': 1,
+          },
+          {
+            'test_case_id': '36340965686c32694f88f06c6a3f71ac',
+            status: 'PASS',
+            'duration_ms': 123,
+            'job_id': 1,
+          },
+          {
+            'test_case_id': 'e3c1257d76c0b4d0f0c1307161ab5424',
+            'status': 'SKIP',
+            'duration_ms': 1234,
+            'job_id': 1,
+          },
+        ];
+
+        expect(testRuns).toMatchObject(sampleTestRuns);
+      });
     });
+  });
 
-    it('does not duplicate test cases', async () => {
-      await testResultRecord.storeTravisReport({
-        job: sampleJob,
-        build: sampleBuild,
-        result: smallerSampleKarmaReport,
+  describe('database fetching', () => {
+    describe('getTestRunsOfBuild', () => {
+      beforeAll(async () => {
```

Comment:
I think you meant to put this beforeAll block into the "database fetching" describe block, not this inner one, so that it works for the other query method tests too

Reply:
Correct! Fixed.